autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/cm/ClouderaManagerTotalHostsEvaluator.java

```diff
@@ -0,0 +1,63 @@
+package com.sequenceiq.periscope.monitor.evaluator.cm;
+
+import com.cloudera.api.swagger.HostsResourceApi;
+import com.cloudera.api.swagger.client.ApiClient;
+import com.sequenceiq.cloudbreak.client.HttpClientConfig;
+import com.sequenceiq.cloudbreak.cm.DataView;
+import com.sequenceiq.cloudbreak.cm.client.ClouderaManagerApiClientProvider;
+import com.sequenceiq.cloudbreak.cm.client.retry.ClouderaManagerApiFactory;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.ClusterManager;
+import com.sequenceiq.periscope.domain.ClusterManagerVariant;
+import com.sequenceiq.periscope.monitor.evaluator.ClusterManagerTotalHostsEvaluator;
+import com.sequenceiq.periscope.service.security.TlsHttpClientConfigurationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+
+@Component("ClouderaManagerTotalHostsEvaluator")
+@Scope("prototype")
+public class ClouderaManagerTotalHostsEvaluator implements ClusterManagerTotalHostsEvaluator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClouderaManagerTotalHostsEvaluator.class);
+
+    @Inject
+    private TlsHttpClientConfigurationService tlsHttpClientConfigurationService;
+
+    @Inject
+    private SecretService secretService;
+
+    @Inject
+    private ClouderaManagerApiClientProvider clouderaManagerApiClientProvider;
+
+    @Inject
+    private ClouderaManagerApiFactory clouderaManagerApiFactory;
+
+    @Override
+    public ClusterManagerVariant getSupportedClusterManagerVariant() {
+        return ClusterManagerVariant.CLOUDERA_MANAGER;
+    }
+
+    @Override
+    public int getTotalHosts(Cluster cluster) {
+        try {
+            Long clusterId = cluster.getId();
+            LOGGER.debug("Checking number of total hosts for cluster {}.", clusterId);
+            HttpClientConfig httpClientConfig = tlsHttpClientConfigurationService.buildTLSClientConfig(cluster.getStackCrn(),
+                    cluster.getClusterManager().getHost(), cluster.getTunnel());
+            ClusterManager cm = cluster.getClusterManager();
+            String user = secretService.get(cm.getUser());
+            String pass = secretService.get(cm.getPass());
+            ApiClient client = clouderaManagerApiClientProvider.getClient(Integer.valueOf(cm.getPort()), user, pass, httpClientConfig);
+            HostsResourceApi hostsResourceApi = clouderaManagerApiFactory.getHostsResourceApi(client);
+            return hostsResourceApi.readHosts(DataView.FULL.name()).getItems().size();
```

Comment:
is it really necessary to ask for full instead of `SUMMARY`?

Reply:
Done.